presto-client/src/main/java/com/facebook/presto/client/ClientSession.java

```diff
@@ -88,6 +92,27 @@ public static ClientSession withProperties(ClientSession session, Map<String, St
                 session.getTimeZoneId(),
                 session.getLocale(),
                 properties,
+                session.getPreparedStatements(),
+                session.getTransactionId(),
+                session.isDebug(),
+                session.getClientRequestTimeout());
+    }
+
+    public static ClientSession withPreparedStatements(ClientSession session, Map<String, String> preparedStatements)
```

Comment:
This method doesn't work, since it merges the existing properties with the supplied properties, but deallocate expects to replace.

I would make the method replace and make sure that works for all callers.

(The existing, confusingly named methods `withProperties` and `withSessionProperties ` should also be converted into a single method that always does replace.)